src/jdk.compiler/share/classes/com/sun/tools/javac/tree/JCTree.java

```diff
@@ -243,6 +243,8 @@ public enum Tag {
         DEFAULTCASELABEL,
         PARENTHESIZEDPATTERN,
 
+        DECONSTRUCTIONPATTERN,
```

Comment:
might want to rename to RECORDPATTERN (but this is impl dependent, so less important to fix)